lib/native/auth.ts

```diff
@@ -29,43 +29,20 @@ export class AwsCredentialsProvider extends crt_native.AwsCredentialsProvider {
     }
 }
 
-export class AwsSigningConfig extends crt_native.AwsSigningConfig {
-    constructor(
-        algorithm = AwsSigningAlgorithm.SigV4Header,
-        provider: AwsCredentialsProvider,
-        region: string,
-        service?: string,
-        date = new Date(),
-        param_blacklist?: string[],
-        use_double_uri_encode = false,
-        should_normalize_uri_path = true,
-        sign_body = true,
-    ) {
-        super(
-            algorithm,
-            provider,
-            region,
-            service,
-            date,
-            param_blacklist,
-            use_double_uri_encode,
-            should_normalize_uri_path,
-            sign_body,
-        );
-    }
-}
+export type AwsSigningConfig = crt_native.AwsSigningConfig;
 
-export class AwsSigner extends crt_native.AwsSigner {
-
-    public async sign_request(request: HttpRequest, config: AwsSigningConfig): Promise<HttpRequest> {
-        return new Promise((resolve, reject) => {
-            super.sign_request(request, config, (error_code) => {
+export async function sign_request_aws(request: HttpRequest, config: AwsSigningConfig): Promise<HttpRequest> {
```

Comment:
naming: sign_aws_request() ???

Reply:
aws_sign_request() ???

Reply:
I like aws_sign_request(). everything else in this file is prefixed "Aws" and it sounds good